docs/application/native/guides/machine-learning/machine-learning-service.md

```diff
@@ -0,0 +1,75 @@
+# Machine Learning Service
+
+The Machine Learning Service API provides utility interfaces for AI application developers.
+
+These functions allow the following operations with NNStreamer:
+- Set and get the pipeline description with a name, which is a unique key string of applicable service.
+- Delete the pipeline description.
+
+## Prerequisites
+
+To enable your application to use Machine Learning Service API:
+
+1. To use the functions and data types of the Machine Learning Service API, include the `<ml-api-service.h>` header file in your application:
+
+   ```c
+   #include <ml-api-service.h>
+   ```
+
+2. To use the Machine Learning Service API, include the following features in your `tizen-manifest.xml` file:
+
+   ```xml
+   <feature name="http://tizen.org/feature/machine_learning">true</feature>
+   <feature name="http://tizen.org/feature/machine_learning.service">true</feature>
+   ```
+
+## An example of usage
+
+The following code snippet shows a simple example of the Machine Learning Service API.
+
+A ML developer adds a pipeline description which includes preprocessing and invoking a neural network model, and an application developer can get the registered pipeline.
+
+1. Set the pipeline description.
+```c
+const gchar my_pipeline[] = "videotestsrc is-live=true ! videoconvert ! tensor_converter ! tensor_sink async=false";
+int status;
+
+status = ml_service_set_pipeline ("my_pipeline", my_pipeline);
+if (status != ML_ERROR_NONE) {
+  // handle error case
+  goto error;
+}
+```
+
+2. Get the pipeline description and construct this for various data-streamed service.
```

Comment:
Code snippets must be preceded by a “colon”, and the word "service" should be plural here. 
```suggestion
2. Get the pipeline description and construct this for various data-streamed services: 
```